src/mokka/src/main/java/pl/hycom/mokka/service/file/DefaultFileService.java

```diff
@@ -0,0 +1,59 @@
+package pl.hycom.mokka.service.file;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ * @author Mariusz Krysztofowicz (mariusz.krysztofowicz@hycom.pl)
+ */
+
+/**
+ * Service responsible for file management, implements FileService interface
+ */
+@Component
+public class DefaultFileService implements FileService {
+    private static final Logger LOG = Logger.getLogger(DefaultFileService.class);
+    /**
+     * Location for files
+     */
+    @Value("${file.download.directory}")
+    private String sourceDirectory;
+
+    /**
+     * Gets file by given fileName contained in configured directory, if file doesn't exist throws FileNotFoundException
+     *
+     * @param fileName
+     *         String Searching file name
+     * @return File
+     * @throws FileNotFoundException
+     */
+    @Override
+    public File fetchFile(String fileName) throws FileNotFoundException {
+        LOG.debug("Invoking DefaultFileService#fetchFile with argument [" + fileName + "]");
+        if (StringUtils.isEmpty(fileName)) {
+            throw new FileNotFoundException("File name cannot be empty");
```

Comment:
Well, in fact it is IllegalArgumentException not FileNotFoundException.
Again, I will encourage you to use what @AdamczykMichal proposed: [https://github.com/google/guava/wiki/PreconditionsExplained](url)
Simple and great solution and you do not have to throw exception on your own.